File: UnidirectionalLinList1.java
// Класс узла однонаправленного списка
class Node {
    int data;        // Данные узла
    Node next;       // Ссылка на следующий узел
    
    // Конструктор узла
    public Node(int data) {
        this.data = data;
        this.next = null;  // По умолчанию следующий узел отсутствует
    }
}

// Класс однонаправленного связного списка
class LinkedList {
    private Node head;  // Ссылка на первый узел списка
    
    // Конструктор списка
    public LinkedList() {
        this.head = null;  // Изначально список пустой
    }
    
    // Добавление элемента в конец списка
    public void append(int data) {
        // Создаем новый узел
        Node newNode = new Node(data);
        
        // Если список пустой, новый узел становится головой
        if (head == null) {
            head = newNode;
            return;
        }
        
        // Ищем последний узел в списке
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        
        // Добавляем новый узел в конец
        current.next = newNode;
    }
    
    // Вывод списка на экран
    public void display() {
        if (head == null) {
            System.out.println("Список пуст!");
            return;
        }
        
        Node current = head;
        System.out.print("Список: ");
        while (current != null) {
            System.out.print(current.data + " -> ");
            current = current.next;
        }
        System.out.println("NULL");
    }
    
    // Получение размера списка
    public int size() {
        int count = 0;
        Node current = head;
        while (current != null) {
            count++;
            current = current.next;
        }
        return count;
    }
}

// Демонстрация работы
public class Main {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        
        // Добавляем элементы в список
        list.append(5);
        list.append(10);
        list.append(15);
        
        // Выводим список
        list.display();
        
        // Показываем размер
        System.out.println("Размер списка: " + list.size());
    }
}

File: UnidirectionalLinList1.cpp
#include <iostream>
using namespace std;

// Структура узла однонаправленного списка
struct Node {
    int data;       // Данные узла
    Node* next;     // Указатель на следующий узел
    
    // Конструктор для удобного создания узлов
    Node(int value) : data(value), next(nullptr) {}
};

class LinkedList {
private:
    Node* head;     // Указатель на первый узел списка

public:
    // Конструктор - инициализируем пустой список
    LinkedList() : head(nullptr) {}
    
    // Добавление элемента в конец списка
    void append(int data) {
        // Создаем новый узел
        Node* newNode = new Node(data);
        
        // Если список пустой - новый узел становится головой
        if (head == nullptr) {
            head = newNode;
            return;
        }
        
        // Идем до последнего узла в списке
        Node* current = head;
        while (current->next != nullptr) {
            current = current->next;
        }
        
        // Присоединяем новый узел к концу списка
        current->next = newNode;
    }
    
    // Вывод списка на экран
    void display() {
        if (head == nullptr) {
            cout << "Список пуст!" << endl;
            return;
        }
        
        Node* current = head;
        cout << "Список: ";
        while (current != nullptr) {
            cout << current->data << " -> ";
            current = current->next;
        }
        cout << "NULL" << endl;
    }
    
    // Деструктор - освобождаем память
    ~LinkedList() {
        Node* current = head;
        while (current != nullptr) {
            Node* next = current->next;
            delete current;
            current = next;
        }
    }
};

// Пример использования
int main() {
    LinkedList list;
    
    // Добавляем элементы в список
    list.append(10);
    list.append(20);
    list.append(30);
    
    // Выводим список
    list.display();
    
    return 0;
}
